virtcontainers/sandbox.go

```diff
@@ -1056,6 +1102,7 @@ func (s *Sandbox) AddInterface(inf *grpc.Interface) (*grpc.Interface, error) {
 	}
 
 	// Add network for vm
+	inf.PciAddr = endpoint.PCIAddr
```

Comment:
AFAICT, this is not enough. The endpoint pci address is determined when qemu plugs it to the guest vm but we are not saving it back to the endpoint data structure. You need something like https://github.com/bergwolf/kata-runtime/commit/2e3fb6f34963b1afe886b30dd6dfaaa2791d6e08 to make it work.

Reply:
@bergwolf https://github.com/bergwolf/kata-runtime/commit/2e3fb6f34963b1afe886b30dd6dfaaa2791d6e08 looks a lot like https://github.com/kata-containers/runtime/pull/534/commits/00b48c9c5d77843fb8c5a75a1ed5cc3fed5a8728, where's the diff.
And yes, I have tested this whole PR, I'm able to add and remove additional network interfaces with `docker connect ...`.

Reply:
Ok, I missed that. Sorry for the noise.